test/parser/parser_spec.coffee

```diff
@@ -0,0 +1,12 @@
+chai = require 'chai'
+expect = chai.expect
+
+parse = require '../../parser/parser'
+outputTokens = require '../example/output_tokens'
+
+describe 'Scanner', ->
+
+  describe 'parsing valid teascript programs', ->
+    describe 'valid teascript program #1', ->
+      context "when test program1.tea's tokens
+               are passed through the scanner", ->
```

Comment:
In the future it would be ideal if we had some sort of testing for partial work so that picking up where you left off is easier. There were a couple of problems that were difficult to track down due to lack of test.